Many many changes

Eneris opened the pull request:
Hi. Please see the changes and consider using them :)

* Udated modules
* Typescript
* Minor changes to data structure

selfisekai committed:
:tada: Add skipFcmRegistration option

Some developers just need a valid gcm token to work with some APIs.

selfisekai committed:
:hammer: Add typescript type declarations

dependabot[bot] committed:
:arrow_up: Bump lodash.merge from 4.6.1 to 4.6.2

Bumps [lodash.merge](https://github.com/lodash/lodash) from 4.6.1 to 4.6.2.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/commits)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump http-proxy from 1.16.2 to 1.18.1

Bumps [http-proxy](https://github.com/http-party/node-http-proxy) from 1.16.2 to 1.18.1.
- [Release notes](https://github.com/http-party/node-http-proxy/releases)
- [Changelog](https://github.com/http-party/node-http-proxy/blob/master/CHANGELOG.md)
- [Commits](https://github.com/http-party/node-http-proxy/compare/1.16.2...1.18.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump ini from 1.3.5 to 1.3.7

Bumps [ini](https://github.com/isaacs/ini) from 1.3.5 to 1.3.7.
- [Release notes](https://github.com/isaacs/ini/releases)
- [Commits](https://github.com/isaacs/ini/compare/v1.3.5...v1.3.7)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump y18n from 3.2.1 to 3.2.2

Bumps [y18n](https://github.com/yargs/y18n) from 3.2.1 to 3.2.2.
- [Release notes](https://github.com/yargs/y18n/releases)
- [Changelog](https://github.com/yargs/y18n/blob/master/CHANGELOG.md)
- [Commits](https://github.com/yargs/y18n/commits)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump handlebars from 4.0.11 to 4.7.7

Bumps [handlebars](https://github.com/wycats/handlebars.js) from 4.0.11 to 4.7.7.
- [Release notes](https://github.com/wycats/handlebars.js/releases)
- [Changelog](https://github.com/handlebars-lang/handlebars.js/blob/master/release-notes.md)
- [Commits](https://github.com/wycats/handlebars.js/compare/v4.0.11...v4.7.7)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump lodash from 4.17.5 to 4.17.21

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.5 to 4.17.21.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.5...4.17.21)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump hosted-git-info from 2.5.0 to 2.8.9

Bumps [hosted-git-info](https://github.com/npm/hosted-git-info) from 2.5.0 to 2.8.9.
- [Release notes](https://github.com/npm/hosted-git-info/releases)
- [Changelog](https://github.com/npm/hosted-git-info/blob/v2.8.9/CHANGELOG.md)
- [Commits](https://github.com/npm/hosted-git-info/compare/v2.5.0...v2.8.9)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump merge from 1.2.0 to 1.2.1

Bumps [merge](https://github.com/yeikos/js.merge) from 1.2.0 to 1.2.1.
- [Release notes](https://github.com/yeikos/js.merge/releases)
- [Commits](https://github.com/yeikos/js.merge/compare/v1.2.0...v1.2.1)

Signed-off-by: dependabot[bot] <support@github.com>

koush committed:
switch from request-promise to axios

koush committed:
axios fixup for binaryu

koush committed:
arraybuffer return

koush committed:
remove redundant arraybufferview

Patrick-Remy committed:
:hammer: Add bundleId option on register

Patrick-Remy committed:
:hammer: Fix linting issues

Patrick-Remy committed:
:bug: Fix #34, escape endpoint form data param

Patrick-Remy committed:
:hammer: Git ignore IDE history files

dependabot[bot] committed:
:arrow_up: Bump tar from 2.2.1 to 2.2.2

Bumps [tar](https://github.com/npm/node-tar) from 2.2.1 to 2.2.2.
- [Release notes](https://github.com/npm/node-tar/releases)
- [Changelog](https://github.com/npm/node-tar/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/node-tar/compare/v2.2.1...v2.2.2)

---
updated-dependencies:
- dependency-name: tar
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump path-parse from 1.0.5 to 1.0.7

Bumps [path-parse](https://github.com/jbgutierrez/path-parse) from 1.0.5 to 1.0.7.
- [Release notes](https://github.com/jbgutierrez/path-parse/releases)
- [Commits](https://github.com/jbgutierrez/path-parse/commits/v1.0.7)

---
updated-dependencies:
- dependency-name: path-parse
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:arrow_up: Bump tmpl from 1.0.4 to 1.0.5

Bumps [tmpl](https://github.com/daaku/nodejs-tmpl) from 1.0.4 to 1.0.5.
- [Release notes](https://github.com/daaku/nodejs-tmpl/releases)
- [Commits](https://github.com/daaku/nodejs-tmpl/commits/v1.0.5)

---
updated-dependencies:
- dependency-name: tmpl
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Eneris committed:
Merge commit '609dad739601413debc85dc58c884edbc0133ddb'

Eneris committed:
Merge commit '44a60e111f1813fbff97095084c43131b2e90ff6'

Eneris committed:
Merge commit '08912f7820b5bdc2b0eebc466712f29bf59ac39f'

Eneris committed:
Merge commit '2b8e7ede9d483e94d376109a6c8fbca93e8d85e0'

Eneris committed:
Merge commit 'b28808b3c5201305f72765a0bbd21deff988fc87'

Eneris committed:
Merge commit '43c33ea457aa977e0d793ef21174c6a32eef1bae'

Eneris committed:
Merge commit '6517b946f81cde0545fb60d9d7f9cd284d929bfa'

Eneris committed:
Merge commit '6bcf56f4f8832ef4c4c8851dc27666d58b66f0eb'

# Conflicts:
#	yarn.lock

Eneris committed:
Merge commit 'e9f8b5cee79a2a3cde1cc7ebeebdf5888e2c3c09'

Eneris committed:
Merge commit 'fc2758af1ec87c7b1e6b590c5a57e30123c4d280'

Eneris committed:
Merge commit '271e0b7d062957f59d302266a418a13fb57bb3c1'

Eneris committed:
upgrade packages

Eneris committed:
Merge commit '8dee225e5d92cebb3cbf966fe8d2e66ccd40ef1e'

Eneris committed:
Convert to TypeScript

Eneris committed:
Fix types

djm181 committed:
Clean up onReady handlers after connect

djm181 committed:
Add prepare script

djm181 committed:
Build improvements

djm181 committed:
Incremental builds

Eneris committed:
Merge pull request #1 from djm181/master

Fix onReady listener leak

Eneris committed:
Make tests work again & Update modules

jimmywarting commented:
👎  to typescript
https://jimmywarting.github.io/you-might-not-need-typescript/

jimmywarting commented:
was it also necessary to bring in another yarn dependency. npm comes built in...

Eneris commented:
@jimmywarting If you don't like it, feel free to use the old version

Eneris committed:
Upgrade dependencies

Eneris committed:
Fix tests again

Eneris committed:
Implement Heartbeat

Eneris committed:
Bump version and create changelog

Eneris committed:
Update README.md

satrik committed:
Replace "console" with "Logger"

"console.log" ignores logLevel

Eneris committed:
Merge pull request #3 from satrik/patch-1

Replace "console" with "Logger"

Eneris committed:
More console logs

Commit:
feat: add `axiosConfig` option to the configuration
refactor: typing adjustments in the `doRegister` and `postRegister` functions

Commit:
refactor: remove `request-promise` dependency and use the `mkdirp` and `rimraf` packages instead of plain `mkdir -p` and `rm -rf` commands

Commit:
refactor: add named export to client.ts

Commit:
docs: use the named export in the example

Eneris committed:
Merge pull request #5 from erikian/feat/proxy

feat: add `axiosProxyConfig` option to the configuration

Eneris committed:
Merge pull request #6 from erikian/refactor/package-json

refactor: remove `request-promise` dependency and use the `mkdirp` and `rimraf` packages instead of plain `mkdir -p` and `rm -rf` commands

Eneris committed:
Merge pull request #7 from erikian/refactor/named-export

refactor: add named export to client.ts

Eneris committed:
v3.1.3

Eneris committed:
Upgrade dependencies

Eneris committed:
Removed unused old scripts and deps

Eneris committed:
v3.1.4

Commit:
fix: clear heartbeat timeouts when the client is destroyed

Eneris committed:
Merge pull request #8 from erikian/fix/heartbeat

fix: clear heartbeat timeouts when the client is destroyed

jjfufu committed:
fix: invalid object destructuring

Eneris committed:
Merge pull request #11 from jjfufu/fix/invalid-object-destructuring

fix: invalid object destructuring

Eneris committed:
Update protobuf.js + minor changes to build steps

Eneris committed:
Migrate to FCM v1
Replace axios with native fetch
Moved to Node v20+
Separated sending and
Minor reformat

Eneris committed:
Removed unmaintaned tests

Eneris committed:
Removed duplicate example - example is already in readme.md

Eneris committed:
Bump protobuf version

Eneris committed:
Update README

Eneris committed:
Conver client private properties to native # privates

Eneris committed:
Bump version

Eneris committed:
Merge pull request #16 from Eneris/fcm-v1

Fcm v1

Eneris committed:
Update README.md

Commit:
Support register FCM using default VAPID

tsightler committed:
Apply suggestions from code review

Co-authored-by: Martin Kalábek <Eneris@users.noreply.github.com>

Commit:
Cleanup unused code

Eneris committed:
Merge pull request #20 from tsightler/master

Support FCM register using default VAPID

Eneris committed:
Bump version & Update dependencies & Add basic example

dgreif committed:
Respect debug config

Eneris committed:
Merge pull request #21 from dgreif/fix-debug-config

Respect debug config

Eneris committed:
Bump version

Eneris committed:
Create custom typed EventEmitter and wrap all listeners into try..catch

dgreif committed:
Remove extraneous console log

Eneris committed:
Merge pull request #23 from dgreif/dg/remove-log

Remove extraneous console log

panther7 committed:
Fixed this-context in setTimeout

Eneris committed:
Merge pull request #24 from panther7/fix-settimeout-context

Fixed this-context in setTimeout

Eneris committed:
Remove all socket listeners before destroy

Eneris committed:
Bump versions

Eneris committed:
Added whenReady promise

Eneris committed:
Upgrade dependencies

Eneris committed:
Split request error handling to network and other

Eneris committed:
Bump version and deps

TheLeoP committed:
End socket connection before destroying it

It seems like the socket is kept open (receiving information), even
after the `destroy`, if this is not done

Eneris committed:
Merge pull request #31 from TheLeoP/fix

End socket connection before destroying it

edualm committed:
fix: properly parse crypto-key and encryption headers in decrypt

The crypto-key header can contain multiple parameters separated by
semicolons, e.g.:
  dh=BASE64KEY; p256ecdsa=BASE64SIG

The previous code used .slice(3) to strip the 'dh=' prefix, but this
left the '; p256ecdsa=...' suffix appended to the DH public key value,
causing ERR_CRYPTO_ECDH_INVALID_PUBLIC_KEY errors when the server
includes a p256ecdsa signature in the header.

Similarly, the encryption header was parsed with .slice(5) which is
fragile and would break if additional parameters were added.

This commit properly parses both headers by splitting on semicolons
and extracting parameters by name.

Eneris committed:
Merge pull request #33 from edualm/fix/decrypt-crypto-key-header-parsing

fix: properly parse crypto-key and encryption headers in decrypt

Eneris committed:
Small decrypt failproofing

Eneris committed:
Upgrade packages

Eneris committed:
v4.3.1
